File: Practica 3/Cola.java
package PilasColas;
public class Cola {
    private int n;
    private int inicio;
    private int fin;
    private long array[];;
    public Cola(int n){
        this.array = new long[n];
        this.inicio=0;
        this.fin=0;
        this.n=n;
    }
    public void insert(long e){
        if(!isFull()){
            this.array[fin]=e;
            this.fin++;
        }
        else{
            System.out.println("Cola Llena!!!");
        }
    }
    public long remove(){
        if(!isEmpty()){
            long n = array[this.inicio];
            this.inicio++;
            return n;
        }
        else{
            System.out.println("Cola Vacia!!!");
        }
        return 0l;
    }
    public long peek(){
        if(!isEmpty()){
            long n = array[this.inicio];
            return n;
        }
        else{
            System.out.println("Pila Vacia!!!");
        }
        return 0l;
    }
    public boolean isEmpty(){
        if(this.fin==inicio){
            return true;
        }
        return false;
    }
    public boolean isFull(){
        if(this.fin==this.n){
            return true;
        }
        return false;
    }
    public void mostrar(){
        for(int i=this.inicio;i<this.n;i++){
            System.out.println(array[i]);
        }
    }
    public static void main(String[] args) {
        Cola c= new Cola(3);
        c.insert(1l);
        c.insert(2l);
        c.insert(3l);
        c.mostrar();
        System.out.println("");
        System.out.println(c.remove());
        System.out.println("");
        c.mostrar();
        System.out.println("");
        System.out.println(c.peek());
        System.out.println("");
        c.mostrar();
    }
    
}

File: Practica 3/Cola.py
class Cola:
    def __init__(self, n):
        self.n = n
        self.inicio = 0
        self.fin = 0
        self.array = [0] * n

    def insert(self, e):
        if not self.isFull():
            self.array[self.fin] = e
            self.fin += 1
        else:
            print("Cola Llena!!!")

    def remove(self):
        if not self.isEmpty():
            n = self.array[self.inicio]
            self.inicio += 1
            return n
        else:
            print("Cola Vacia!!!")
        return 0

    def peek(self):
        if not self.isEmpty():
            return self.array[self.inicio]
        else:
            print("Cola Vacia!!!")
        return 0

    def isEmpty(self):
        return self.fin == self.inicio

    def isFull(self):
        return self.fin == self.n

    def mostrar(self):
        for i in range(self.inicio, self.fin):
            print(self.array[i])

if __name__ == "__main__":
    c = Cola(3)
    c.insert(1)
    c.insert(2)
    c.insert(3)
    c.mostrar()
    print("")
    print(c.remove())
    print("")
    c.mostrar()
    print("")
    print(c.peek())
    print("")
    c.mostrar()
